### ISSUE TITLE
Previous day's consumption included in first hour of new day consumption

### ISSUE THREAD

**Issue by Quickstep3** [BottlecapDave/HomeAssistant-OctopusEnergy#726]
### Describe the bug

Bug in that the previous day's total usage is re-report/included on the 00:00-01:00 hour of the following day.
This only appears to happening with gas usage, but has happened at least twice. There is definitely no actual usage between 00:00-01:00 as the heating is off and Octopus app confirms no usage at that point as well. Only change was that I upgrade to the version of the integration during the day yesterday.
Today's (27/01/2024) Gas usage reporting: 
<img width="771" alt="image" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/72910606/3475b387-9904-40e9-b44e-868d16131773">
Prior day's usage (26/01/2024). Note total Gas import in distribution and lack of any usage overnight (heating is off)
<img width="1064" alt="image" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/72910606/fb21ae68-028a-4b88-8979-e5b1c3c8598f">

Previously happened on 1st-2nd January but that was reported as negative import of the previous days' total usage for gas so likely different problem (will have been different version of integration too)

History logs from Gas meter show the following:
<img width="1057" alt="image" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/72910606/863b13ac-f15c-4d38-9121-f2cd05e82c71">
So appears that something didn't roll over at midnight and took until 00:15ish to update?

### Reproduction steps

Unsure - nothing obvious in logs 

### Expected behaviour

Gas usage reporting is not duplicated

### Tariff Code

G-1R-SILVER-FLEX-22-11-25-H

### Integration Version

10.0.3

### Home Assistant Version

2024.1.5

### Fresh Install?

After upgrading

### Home Assistant Logs

{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2024.1.5",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.11.6",
    "docker": true,
    "arch": "aarch64",
    "timezone": "Europe/London",
    "os_name": "Linux",
    "os_version": "6.1.63-haos-raspi",
    "supervisor": "2023.12.1",
    "host_os": "Home Assistant OS 11.4",
    "docker_version": "24.0.7",
    "chassis": "embedded",
    "run_as_root": true
  },
  "custom_components": {
    "hacs": {
      "version": "1.34.0",
      "requirements": [
        "aiogithubapi>=22.10.1"
      ]
    },
    "solaredge_modbus_multi": {
      "version": "2.4.8",
      "requirements": [
        "pymodbus>=3.5.4"
      ]
    },
    "kia_uvo": {
      "version": "2.18.2",
      "requirements": [
        "hyundai_kia_connect_api==3.11.4"
      ]
    },
    "hypervolt_charger": {
      "version": "1.3.1",
      "requirements": [
        "websockets==11.0.3"
      ]
    },
    "localtuya": {
      "version": "5.2.1",
      "requirements": []
    },
    "octopus_energy": {
      "version": "10.0.3",
      "requirements": []
    },
    "solaredge_modbus": {
      "version": "1.11.1",
      "requirements": [
        "pymodbus==3.5.4"
      ]
    }
  },
  "integration_manifest": {
    "domain": "octopus_energy",
    "name": "Octopus Energy",
    "codeowners": [
      "@bottlecapdave"
    ],
    "config_flow": true,
    "dependencies": [
      "repairs",
      "recorder"
    ],
    "documentation": "https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy",
    "homekit": {},
    "iot_class": "cloud_polling",
    "issue_tracker": "https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues",
    "ssdp": [],
    "version": "10.0.3",
    "zeroconf": [],
    "is_built_in": false
  },
  "data": {
    "id": "A-64C14B36",
    "octoplus_enrolled": true,
    "electricity_meter_points": [
      {
        "mpan": "**REDACTED**",
        "meters": [
          {
            "serial_number": "**REDACTED**",
            "is_export": false,
            "is_smart_meter": true,
            "device_id": "**REDACTED**",
            "manufacturer": "1057 - Honeywell Elster",
            "model": "00000101",
            "firmware": "09014009"
          }
        ],
        "agreements": [
          {
            "start": "2023-10-24T23:00:00+00:00",
            "end": null,
            "tariff_code": "E-1R-GO-VAR-22-10-14-H",
            "product_code": "GO-VAR-22-10-14"
          }
        ]
      },
      {
        "mpan": "**REDACTED**",
        "meters": [
          {
            "serial_number": "**REDACTED**",
            "is_export": true,
            "is_smart_meter": true,
            "device_id": null,
            "manufacturer": "1057 - Honeywell Elster",
            "model": "00000101",
            "firmware": "09014009"
          }
        ],
        "agreements": [
          {
            "start": "2023-10-24T23:00:00+00:00",
            "end": "2024-10-24T23:00:00+00:00",
            "tariff_code": "E-1R-OUTGOING-LITE-FIX-12M-23-09-12-H",
            "product_code": "OUTGOING-LITE-FIX-12M-23-09-12"
          }
        ]
      }
    ],
    "gas_meter_points": [
      {
        "mprn": "**REDACTED**",
        "meters": [
          {
            "serial_number": "**REDACTED**",
            "consumption_units": "m\u00b3",
            "is_smart_meter": true,
            "device_id": "**REDACTED**",
            "manufacturer": "1192 - George Wilson",
            "model": "11111111",
            "firmware": "01C800D1"
          }
        ],
        "agreements": [
          {
            "start": "2023-02-15T00:00:00+00:00",
            "end": null,
            "tariff_code": "G-1R-SILVER-FLEX-22-11-25-H",
            "product_code": "SILVER-FLEX-22-11-25"
          }
        ]
      }
    ]
  }
}

### Confirmation

- [X] I confirm that I cannot find my solution within the [documentation](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy)
- [X] I confirm that I cannot find my solution within the [FAQ](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/faq)

**Comment by jksj461** [comment]
29/01/24 Just to add, I am seeing the same issue on Gas Current consumption. My electricity meter is currently broken so I cant confirm the issue on both sensors. It had previously happened a few times earlier in the month after the upgrade to version 10.0.  I just assumed it was a glitch and had not twigged it was the Previous Days Consumption. The issue is as described above.

**Comment by BottlecapDave** [comment]
@jksj461 as mentioned above. Please provide logs around problem time so I can try and work out what's gone wrong.

**Comment by smsm1** [comment]
I've had this issue too for the electricity meter on 8 Feb 2024 and 12 Feb 2024. Gas had the issue on 12 Feb 2024. I've enabled debug log for if the issue happens again.

Here's some potentially relevant warning in the home assistant core log:

```
2024-02-12 10:27:02.189 WARNING (SyncWorker_2) [homeassistant.loader] We found a custom integration octopusagile which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-02-12 10:27:02.207 WARNING (SyncWorker_2) [homeassistant.loader] We found a custom integration octopus_energy which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-02-12 10:27:54.565 WARNING (MainThread) [custom_components.octopus_energy.coordinators.intelligent_dispatches] Failed to retrieve new dispatches. Next attempt at 2024-02-12 10:28:54.534456+00:00
2024-02-12 10:27:54.573 WARNING (MainThread) [custom_components.octopus_energy.coordinators.intelligent_settings] Failed to retrieve new intelligent settings. Next attempt at 2024-02-12 10:28:54.566070+00:00
2024-02-12 10:27:56.280 WARNING (MainThread) [homeassistant.const] ENERGY_KILO_WATT_HOUR was used from octopus_energy, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfEnergy.KILO_WATT_HOUR instead, please create a bug report at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues
2024-02-12 10:27:56.309 WARNING (MainThread) [homeassistant.const] ENERGY_KILO_WATT_HOUR was used from octopus_energy, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfEnergy.KILO_WATT_HOUR instead, please create a bug report at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues
2024-02-12 10:27:56.328 WARNING (MainThread) [homeassistant.const] ENERGY_KILO_WATT_HOUR was used from octopus_energy, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfEnergy.KILO_WATT_HOUR instead, please create a bug report at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues
2024-02-12 10:27:56.348 WARNING (MainThread) [homeassistant.const] ENERGY_KILO_WATT_HOUR was used from octopus_energy, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfEnergy.KILO_WATT_HOUR instead, please create a bug report at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues
2024-02-12 10:27:56.384 WARNING (MainThread) [homeassistant.const] ENERGY_KILO_WATT_HOUR was used from octopus_energy, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfEnergy.KILO_WATT_HOUR instead, please create a bug report at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues
2024-02-12 10:27:56.405 WARNING (MainThread) [homeassistant.const] ENERGY_KILO_WATT_HOUR was used from octopus_energy, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfEnergy.KILO_WATT_HOUR instead, please create a bug report at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues
2024-02-12 10:27:56.419 WARNING (MainThread) [homeassistant.const] VOLUME_CUBIC_METERS was used from octopus_energy, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfVolume.CUBIC_METERS instead, please create a bug report at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues
2024-02-12 10:27:56.470 WARNING (MainThread) [homeassistant.const] ENERGY_KILO_WATT_HOUR was used from octopus_energy, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfEnergy.KILO_WATT_HOUR instead, please create a bug report at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues
2024-02-12 10:27:56.491 WARNING (MainThread) [homeassistant.const] ENERGY_KILO_WATT_HOUR was used from octopus_energy, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfEnergy.KILO_WATT_HOUR instead, please create a bug report at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues
2024-02-12 10:27:56.552 WARNING (MainThread) [homeassistant.const] VOLUME_CUBIC_METERS was used from octopus_energy, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfVolume.CUBIC_METERS instead, please create a bug report at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues
2024-02-12 10:27:56.568 WARNING (MainThread) [homeassistant.const] ENERGY_KILO_WATT_HOUR was used from octopus_energy, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfEnergy.KILO_WATT_HOUR instead, please create a bug report at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues
2024-02-12 10:27:56.588 WARNING (MainThread) [homeassistant.const] ENERGY_KILO_WATT_HOUR was used from octopus_energy, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfEnergy.KILO_WATT_HOUR instead, please create a bug report at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues
2024-02-12 10:27:56.604 WARNING (MainThread) [homeassistant.const] ENERGY_KILO_WATT_HOUR was used from octopus_energy, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfEnergy.KILO_WATT_HOUR instead, please create a bug report at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues
2024-02-12 10:30:10.319 WARNING (Recorder) [homeassistant.components.sensor.recorder] The unit of sensor.octopus_energy_gas_e6p20482092200_1078351705_current_rate (GBP/kWh) cannot be converted to the unit of previously compiled statistics (None). Generation of long term statistics will be suppressed unless the unit changes back to None or a compatible unit. Go to https://my.home-assistant.io/redirect/developer_statistics to fix this
2024-02-12 19:58:13.240 WARNING (MainThread) [homeassistant.helpers.entity] Update of sensor.octopus_energy_gas_e6p20482092200_1078351705_previous_accumulative_consumption_kwh is taking over 10 seconds
2024-02-12 19:58:13.241 WARNING (MainThread) [homeassistant.helpers.entity] Update of sensor.octopus_energy_electricity_22j0423673_1800018868183_previous_accumulative_cost is taking over 10 seconds
2024-02-13 00:12:53.461 WARNING (MainThread) [custom_components.octopus_energy.api_client] Failed to connect. Timeout of ClientTimeout(total=15, connect=None, sock_read=None, sock_connect=None, ceil_threshold=5) exceeded.
2024-02-13 00:12:53.462 WARNING (MainThread) [custom_components.octopus_energy.coordinators.current_consumption] Failed to retrieve smart meter consumption data - using cached version. Next attempt at 2024-02-13 00:13:34.814612+00:00
```

Unfortunately I can't see further back in the log ui as there had been a reboot yesterday.

I've just seen in HACS there's an update for Octopus Energy from 9.2.1 to 10.1.0, so it's possible that could fix things. Looking at the change log and one of the graphs someone else posted it's possible there's a fix in 10.0.3 for this.

**Comment by mitchellrj** [comment]
This is more than likely an issue with the meter itself or the GPF. I've worked with high volume smart meter data and this sort of thing crops up a lot, particularly with gas.
